raphtory/src/core/mod.rs

```diff
@@ -37,6 +45,65 @@ pub mod state;
 pub(crate) mod storage;
 pub mod utils;
 
+/// this is here because Arc<str> annoyingly doesn't implement all the expected comparisons
+#[derive(Clone, Debug, Eq, Ord, Hash, Serialize, Deserialize)]
+pub struct ArcStr(pub(crate) Arc<str>);
+
+impl Display for ArcStr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        Display::fmt(&self.0, f)
+    }
+}
+
+impl<T: Into<Arc<str>>> From<T> for ArcStr {
+    fn from(value: T) -> Self {
+        ArcStr(value.into())
+    }
+}
+
+impl From<ArcStr> for String {
+    fn from(value: ArcStr) -> Self {
+        value.to_string()
+    }
+}
+impl Deref for ArcStr {
+    type Target = Arc<str>;
```

Comment:
Isn't this `str` ?

Reply:
My thinking was that it is wrapping an `Arc<str>` so you should be able to do anything that you can do with an Arc. Derek coercion should chain the calls to `deref()` for you so this should still act like it implements `Deref<Targe=str>` since that is implemented for `Arc<str>`.

Reply:
fair